### PR TITLE
ci: use new image with gcc-7

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

*Description*:
Use latest build-image with gcc-7 and limit resources not only by CPU.

*Risk Level*: Low
*Testing*: CI
*Docs Changes*:
*Release Notes*:

### CODE REVIEW

**Comment by lizan** [comment]
Yeah I'm trying to see what CI outcome is by this PR. We will see...

**Comment by lizan** [comment]
@mattklein123 yeah, but the error indicates different issue than memory, I'm looking into it..

File: .bazelrc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Envoy specific Bazel build/test options.
-
+startup --host_jvm_args=-Xmx2048m
```
Comment (mattklein123) [inline .bazelrc]: How does this help if Bazel allocates memory above this? Won't it still OOM? Or does this force GC more often as it approaches the limit? Can you add a comment on how this was selected?

File: .circleci/config.yml [hunk 2]
```diff
@@ -14,7 +14,10 @@ jobs:
      steps:
        - run: rm -rf /home/circleci/project/.git # CircleCI git caching is likely broken
        - checkout
-       - run: ci/do_circle_ci.sh bazel.release
+       - run:
+           command:
+             ci/do_circle_ci.sh bazel.release
+           no_output_timeout: 60m
```
Comment (mattklein123) [inline .circleci/config.yml]: Why is this needed here and below now? Shouldn't Bazel still be outputing status?

File: ci/do_ci.sh [hunk 3]
```diff
@@ -89,8 +89,8 @@ if [[ "$1" == "bazel.release" ]]; then
     # https://github.com/envoyproxy/envoy/pull/5611.
     # TODO(akonradi): use --local_cpu_resources flag once Bazel has a release
     # after 0.21.
-    [ -z "$CIRCLECI" ] || export BAZEL_BUILD_OPTIONS="${BAZEL_BUILD_OPTIONS} --local_resources=12288,6,1"
-    [ -z "$CIRCLECI" ] || export BAZEL_TEST_OPTIONS="${BAZEL_TEST_OPTIONS} --local_resources=12288,6,1"
+    [ -z "$CIRCLECI" ] || export BAZEL_BUILD_OPTIONS="${BAZEL_BUILD_OPTIONS} --local_resources=12288,5,1"
```
Comment (mattklein123) [inline ci/do_ci.sh]: I'm not sure how much of a speedup it would be, but I'm still pretty sure you could build `//source` with full CPUs, and then proceed to build the tests with less.

File: .bazelrc (hunk 1 above)
Comment (lizan) [reply .bazelrc]: No it won't, this is to control GC, added comment.

File: .bazelrc [hunk 4]
```diff
@@ -1,5 +1,10 @@
 # Envoy specific Bazel build/test options.
 
+# Bazel doesn't need less than 200MB of memory based on memroy profiling:
```
Comment (ggreenway) [inline .bazelrc]: spelling: memroy

Comment (ggreenway) [reply .bazelrc]: Also, this sentence is hard to parse. It reads to me as "bazel does need more than 200MB", but I'm not sure that's what you meant.

Comment (lizan) [reply .bazelrc]: thanks! yes that what I meant 🤦‍♂️

File: .bazelrc (hunk 1 above)
Comment (mattklein123) [reply .bazelrc]: If it doesn't need more than 200MB, why are we giving it 2GB? To account for GC slop?

**Comment by mattklein123** [comment]
@lizan this still seems to be failing coverage?

File: .bazelrc (hunk 1 above)
Comment (lizan) [reply .bazelrc]: I don't have specific reason for that, let's try lower number :)

**Comment by lizan** [comment]
hmm, I couldn't reproduce coverage failure locally with docker memory limit (could repro with 6+ CPU resources)...

File: .circleci/config.yml (hunk 2 above)
Comment (mattklein123) [reply .circleci/config.yml]: @htuch any chance to let the G internal gtest folks know about this apparent perf degradation?

Comment (htuch) [reply .circleci/config.yml]: Yes, but can we quantify somehow?

**Comment by lizan** [comment]
@htuch @mattklein123 Last 3 commit passes coverage / release (asan failure is irrelevant), I think we can merge this once 4th one passes. Needs your approval again :)
